hudi-common/src/main/java/org/apache/hudi/avro/AvroRecordSizeEstimator.java

```diff
@@ -45,6 +49,8 @@ public long sizeEstimate(BufferedRecord<IndexedRecord> record) {
       return sizeOfRecord;
     }
     // do not contain size of Avro schema as the schema is reused among records
+    Schema recordSchema = record.getRecord().getSchema();
+    long sizeOfSchema = sizeOfSchemaMap.computeIfAbsent(recordSchema.getFields().size(), arity -> ObjectSizeCalculator.getObjectSize(recordSchema));
```

Comment:
cc @cshuo to review this, guess there is no much perf regression after the change?